sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -0,0 +1,290 @@
+    @abc.abstractmethod
+    def algorithm_exogenous_support(self: "_NeuralForecastAdapter") -> bool:
+        """Set support for exogenous features."""
+
+    @functools.cached_property
+    @abc.abstractmethod
+    def algorithm_name(self: "_NeuralForecastAdapter") -> str:
+        """Set custom model name."""
+
+    @functools.cached_property
+    @abc.abstractmethod
+    def algorithm_class(self: "_NeuralForecastAdapter"):
+        """Import underlying NeuralForecast algorithm class."""
+
+    @functools.cached_property
+    @abc.abstractmethod
+    def algorithm_parameters(self: "_NeuralForecastAdapter") -> dict:
+        """Get keyword parameters for the underlying NeuralForecast algorithm class.
+
+        Returns
+        -------
+        dict
+            keyword arguments for the underlying algorithm class
+
+        Notes
+        -----
+        This method should not include following parameters:
+
+        - future exogenous columns (``futr_exog_list``) - used from ``__init__``
+        - historical exogenous columns (``hist_exog_list``) - not supported
+        - statis exogenous columns (``stat_exog_list``) - not supported
+        - custom model name (``alias``) - used from ``algorithm_name``
+        """
+
+    def _instantiate_model(self: "_NeuralForecastAdapter", fh: ForecastingHorizon):
+        """Instantiate the model."""
+        exogenous_parameters = (
+            {"futr_exog_list": self.futr_exog_list} if self.needs_X else {}
+        )
+
+        algorithm_instance = self.algorithm_class(
+            fh,
+            alias=self.algorithm_name,
+            **self.algorithm_parameters,
+            **exogenous_parameters,
+        )
+
+        from neuralforecast import NeuralForecast
+
+        model = NeuralForecast(
+            [algorithm_instance], self.freq, local_scaler_type=self.local_scaler_type
+        )
+
+        return model
+
+    def _fit(
+        self: "_NeuralForecastAdapter",
+        y: pandas.Series,
+        X: typing.Optional[pandas.DataFrame],
+        fh: ForecastingHorizon,
+    ) -> "_NeuralForecastAdapter":
+        """Fit forecaster to training data.
+
+        private _fit containing the core logic, called from fit
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        y : sktime time series object
+            guaranteed to have a single column/variable
+        fh : guaranteed to be ForecastingHorizon
+            The forecasting horizon with the steps ahead to to predict.
+        X : sktime time series object, optional (default=None)
+            guaranteed to have at least one column/variable
+            Exogeneous time series to fit to.
+
+        Returns
+        -------
+        self : _NeuralForecastAdapter
+            reference to self
+        """
+        if not fh.is_all_out_of_sample(cutoff=self.cutoff):
+            raise NotImplementedError("in-sample prediction is currently not supported")
```

Comment:
I have this check in `_fit` already, and `fh` passed to `_predict` is unused. Cna you please tell what do you want me to change?

> the in-sample failures can be silenced by raising NotImplementedError if not fh_is_all_in_sample(). @benHeid had the same issue recently, probably we should do that in the base class.

I am confused a bit, why check for `if not fh_is_all_in_sample()`?

Reply:
you seem to be skipping the in-sample tests in `tests/_config` - that should be unnecessary if you raise this error.

Reply:
I think I tested locally earlier and it failed, but didn't check if that skip is necessary after these lines. Will remove that skip from there.